src/instruction.rs

```diff
@@ -0,0 +1,422 @@
+use lazy_static::lazy_static;
+
+#[derive(Debug)]
+enum Mode {
+    Immediate,
+    ZeroPage,
+    ZeroPageX,
+    ZeroPageY,
+    Absolute,
+    AbsoluteX,
+    AbsoluteY,
+    Indirect,
+    IndirectX,
+    IndirectY,
+    Implicit,
+    Accumulator
+}
+
+//Temporary
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct Instruction {
+    name: &'static str,
+    opcode: u8,
+    mode: Mode,
+    len: u8
+}
+const ADC_SET: [Instruction; 8] = [
+    Instruction{name: "ADC", opcode: 0x69, mode: Mode::Immediate,   len: 2},
+    Instruction{name: "ADC", opcode: 0x65, mode: Mode::ZeroPage,    len: 2},
+    Instruction{name: "ADC", opcode: 0x75, mode: Mode::ZeroPageX,   len: 2},
+    Instruction{name: "ADC", opcode: 0x6D, mode: Mode::Absolute,    len: 3},
+    Instruction{name: "ADC", opcode: 0x7D, mode: Mode::AbsoluteX,   len: 3},
+    Instruction{name: "ADC", opcode: 0x79, mode: Mode::AbsoluteY,   len: 3},
+    Instruction{name: "ADC", opcode: 0x61, mode: Mode::IndirectX,   len: 2},
+    Instruction{name: "ADC", opcode: 0x71, mode: Mode::IndirectY,   len: 2},
+];
```

Comment:
how are the instructions categorized? sometimes I see that they have the same `name`, but sometimes it's not the case. Are the categories defined in the spec?

Reply:
ok I stopped being lazy and looked it up and I see that it's from the spec :D, what I find weird is that some of them are not defined in terms of sets (e.g. BRK), I think I'd rather have single item arrays just for consistency,,, but it doesnt really matter lol

Reply:
I can do that for sure. Np